src/library/scala/collection/IterableOnce.scala

```diff
@@ -940,12 +951,24 @@ trait IterableOnceOps[+A, +CC[_], +C] extends Any { this: IterableOnce[A] =>
    */
   def sum[B >: A](implicit num: Numeric[B]): B =
     knownSize match {
-      case -1 => reduceLeftIterator[B](num.zero)(num.plus)
+      case -1 =>
+        val z = num.zero
+        if ((num eq Numeric.BigDecimalIsFractional) || (num eq Numeric.BigDecimalAsIfIntegral)) {
```

Comment:
Er ... this is not Scala.js-friendly. It will *work*, but it will cause a huge code size bump for anyone using `sum` if they don't otherwise use big numbers. That's because `num` now transitively reaches the `java.math.*` big numbers implementations. :(

Reply:
Aha. I'm still open for suggestions, including reverting and documenting the risk of producing an iterator just for isEmpty.

Is it sufficient to push into the Numeric instance? Except for the caveat that "someone somewhere relies on the broken behavior", it may be OK that `+` is ordinary plus and `num.plus` is "safe" plus.